server/eventlog/eventlog.go

```diff
@@ -421,25 +424,31 @@ type WriteWithTailCloser interface {
 type ANSICursorBufferWriter struct {
 	WriteWithTailCloser
 	terminalWriter *terminal.ScreenWriter
-
-	// Number of lines to keep in the screen buffer so that they may be modified
-	// by ANSI Cursor control codes.
-	numLinesToRetainForANSICursor int
 }
 
 func (w *ANSICursorBufferWriter) Write(ctx context.Context, p []byte) (int, error) {
 	if len(p) == 0 {
 		return w.WriteWithTailCloser.WriteWithTail(ctx, p, nil)
 	}
-
 	if _, err := w.terminalWriter.Write(p); err != nil {
 		return 0, err
 	}
-	popped := w.terminalWriter.PopExtraLines(w.numLinesToRetainForANSICursor)
-	if len(popped) != 0 {
-		popped = append(popped, '\n')
+	if w.terminalWriter.ScrollOutFunc == nil {
+		// There's no window giving us scrollOut, this isn't using curses.
```

Comment:
is there some other way to convey this information rather than the nilness of `ScrollOutFunc` (or to push this code and the accompanying `ScrollOut` logic into `terminalWriter`?

Reply:
Yeah, I'll shove this back into the `terminalWriter`. You're right that this logic shouldn't be here.